src/util/simplify_expr_struct.cpp

```diff
@@ -208,6 +209,17 @@ bool simplify_exprt::simplify_member(exprt &expr)
       }
     }
   }
+  else if(op.id() == ID_typecast)
+  {
+    // Try to look through member(cast(x)) if the cast is between structurally
+    // identical types:
+    if(base_type_eq(op_type, op.op0().type(), ns))
+    {
+      expr.op0() = op.op0();
+      simplify_rec(expr);
```

Comment:
I don't think it's necessary to simplify recursively, I'd say a call to `simplify_member` should do the trick.

Reply:
I think this is the only bit keeping me from approving.